test/ProtocolGovernance.spec.ts

```diff
@@ -0,0 +1,307 @@
+        });
+
+        describe("when remove called twice", () => {
+            it("removes the addresses", async () => {
+                
+            });
+        });
+
+        describe("when remove called twice on the same address", () => {
+            it("removes the address and does not fail then", async () => {
+                
+            });
+        });
+
+        describe("when callen by not admin", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+    });
+
+    describe("#removeFromVaultGovernances", () => {
+        describe("when called by not admin", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+
+        describe("when address is not in vault governances", () => {
+            it("does not fail", async () => {
+                
+            });
+        });
+
+        describe("when address is a vault governance", () => {
+            describe("when attempt to remove one address", () => {
+                it("removes", async () => {
+                    
+                });
+            });
+            describe("when attempt to remove multiple addresses", () => {
+                it("removes", async () => {
+                    
+                });
+            });
+        });
+    });
+
+    describe("#setPendingTokenWhitelistAdd", () => {
+        it("does not allow stranger to set pending token whitelist", async () => {
+            
+        });
+
+        it("sets pending token whitelist add and timestamp", async () => {
+            
+        });
+    });
+
+    describe("#commitTokenWhitelistAdd", () => {
+        it("commits pending token whitelist", async () => {
+            
+        });
+
+        describe("when called noy by admin", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+
+        describe("when setPendingTokenWhitelistAdd has not been called", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+
+        describe("when governance delay has not passed or has almost passed", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+
+        describe("when setting to identic addresses", () => {
+            it("passes", async () => {
+                
+            });
+        });
+    });
+
+    describe("#removeFromTokenWhitelist", () => {
+        describe("when called not by admin", () => {
+            it("reverts", async () => {
+                
+            });
+        });
+
+        describe("when passed an address which is not in token whitelist", () => {
+            it("passes", async () => {
+                
+            });
+        });
+
+        it("removes", async () => {
+            
+        });
+
+        describe("when call commit on removed token", () => {
+            it("passes", async () => {
+                
+            });
+        });
```

Comment:
let's add more edge cases 
for example, attempt to remove token that hadn't been whitelisted previously

Reply:
This edge case has already been implemented